operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/DefaultEventHandler.java

```diff
@@ -119,24 +125,17 @@ private EventMonitor monitor() {
 
   @Override
   public void handleEvent(Event event) {
-
     try {
       lock.lock();
-
+      log.debug("Received event: {}", event);
       if (!this.running) {
         log.debug("Skipping event: {} because the event handler is shutting down", event);
         return;
       }
-
-      log.debug("Received event: {}", event);
-
-      final Predicate<CustomResource> selector = event.getCustomResourcesSelector();
       final var monitor = monitor();
-      for (String uid : eventSourceManager.getLatestResourceUids(selector)) {
-        eventBuffer.addEvent(uid, event);
-        monitor.processedEvent(uid, event);
-        executeBufferedEvents(uid);
-      }
+      eventBuffer.addEvent(event.getRelatedCustomResourceID(), event);
+      monitor.processedEvent(event.getRelatedCustomResourceID(), event);
```

Comment:
I know that it was like that even before but wonder if that's correct as in fact here the event is added to a buffer for late processing. In case of failure, the event may also be put back to the buffer so I guess, `processedEvent` should be invoked only when the event has been delivered to the controller. Eventually we can also add metrics about how long it takes for an event from the generation to the delivery. 

@metacosm am I missing something ?

Reply:
That is probably a good point. The event is not processed at this time. We should probably take a look on this in a separate PR.
Note that we plan to remove the event from the context:
https://github.com/java-operator-sdk/java-operator-sdk/issues/538

so not sure how this part will look like after this issue. We might still receive events, but the whole event handling should be simplified.